### PR TITLE
Auth and errors

### PR DESCRIPTION
**Pull request by GBowman1**
## Type of Change
- [x] New Feature
- [ ] Debugging
- [ ] Refactor
- [ ] Documentation

## PR Issue link if applicable
KAN-43, KAN-48 On Jira

## Describe Changes Made
I seemed to have found a solution to handling auth and sessions, everything is working as intended on postman as of now. Also i went back and added some lines for handling more errors in earlier actions. 
Things to not in the future we need to figure out how to handle passing CSRF tokens back and forth with the frontend probably not a good call to exempt our actions in the long run and bring down the security.
Also I attempted sharing my Postman collection so you can see how happy path requests and responses were working as of now.

## PR Checklist
- [x] Added Reviewer
- [ ] Followed TDD, And Test are Passing

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -4,23 +4,43 @@
...
 from django.http import JsonResponse, HttpResponseBadRequest
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.http import require_POST, require_GET
 
 @csrf_exempt
 def create_user(request):
     if request.method == 'POST':
-        data = json.loads(request.body)
+        try:
```
Comment (zach-bergman) [inline api/views.py:14]: What does 'try:' do here?

**Review by zach-bergman — APPROVED** [review]
Thanks for all of this work, I'll have to redo the tests I've written so far but that's no problem. Also the CI is failing here, because of the same test failing twice. Can we merge still and I will redo tests?

File: api/views.py (hunk 1 above)
Comment (GBowman1) [reply api/views.py:14]: its like error handling in rails
it looks like JS tho
try:
do this code (if it errors)
exception(an error catch)
return this error json

**Comment by GBowman1** [comment]
yea we can still merge but it wont deploy till ci passes

File: api/views.py [hunk 2]
```diff
@@ -4,23 +4,43 @@
...
-            'data': {
-                'user_id': user.id,
-                'username': user.username,
-                'email': user.email,
-            }
-        })
+        password = data.get('password')
+        
+        if not username or not email or not password:
+            return JsonResponse({'error': 'Missing required fields'}, status=400)
```
Comment (RodrigoACG) [inline api/views.py:24]: I really like this sad path Logic good job !!

**Review by RodrigoACG — APPROVED** [review]

**Pull request merged**